utils/agent_display_web_with_prompt.py

```
@@ -23,67 +29,67 @@ def __init__(self):
+                raise HTTPException(status_code=500, detail=f"Error reading prompt: {e}")
 
-        @self.app.route('/download/<filename>')
-        def download_file(filename):
+        @self.app.get('/download/{filename}')
+        async def download_file(filename: str):
             try:
-                return send_from_directory(LOGS_DIR, filename, as_attachment=True)
+                file_path = LOGS_DIR / filename
+                if not file_path.exists():
+                    raise HTTPException(status_code=404, detail="File not found")
+                return FileResponse(file_path, filename=filename)
```

Ruff (0.8.2)

Undefined name `FileResponse`

(F821)